### PR TITLE
Add headers and status code to telemetry events

### LINKED ISSUE #80: Telemetry event [:finch, :response, :stop] doesn't include the response status
It would be nice to have the response HTTP status code in the `[:finch, :response, :stop]` event.

### PR DESCRIPTION
**Pull request by dvic**
Fixes #80

### CODE REVIEW

**Comment by dvic** [comment]
I'm working on an open telemetry library for Finch and  I need these fields :) (headers are [optional](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers) according to the spec)

File: lib/finch/http1/conn.ex [hunk 1]
```diff
@@ -175,7 +176,8 @@ defmodule Finch.Conn do
 
   defp handle_response(response, conn, metadata, start_time, extra_measurements) do
     case response do
-      {:ok, mint, acc} ->
+      {:ok, mint, {status, headers, _} = acc} ->
+        metadata = metadata |> Map.put(:status, status) |> Map.put(:headers, headers)
```
Comment (sneako) [inline lib/finch/http1/conn.ex:180]:
```suggestion
        metadata = Map.merge(metadata, %status: status, headers: headers})
```

File: lib/finch/http2/pool.ex [hunk 2]
```diff
@@ -57,7 +58,8 @@ defmodule Finch.HTTP2.Pool do
         result = response_waiting_loop(acc, fun, ref, monitor, fail_safe_timeout)
 
         case result do
-          {:ok, _} ->
+          {:ok, {status, headers, _}} ->
+            metadata = metadata |> Map.put(:status, status) |> Map.put(:headers, headers)
```
Comment (sneako) [inline lib/finch/http2/pool.ex:62]:
```suggestion
            metadata = Map.merge(metadata, %status: status, headers: headers})
```

**Comment by sneako** [comment]
Thank you @dvic !

File: lib/finch/http1/conn.ex [hunk 3]
```diff
@@ -175,7 +176,8 @@ defmodule Finch.Conn do
 
   defp handle_response(response, conn, metadata, start_time, extra_measurements) do
     case response do
-      {:ok, mint, acc} ->
+      {:ok, mint, {status, headers, _} = acc} ->
+        metadata = Map.merge(metadata, %status: status, headers: headers})
```
Comment (sneako) [inline lib/finch/http1/conn.ex:180]:
```suggestion
        metadata = Map.merge(metadata, %{status: status, headers: headers})
```

File: lib/finch/http2/pool.ex [hunk 4]
```diff
@@ -57,7 +58,8 @@ defmodule Finch.HTTP2.Pool do
         result = response_waiting_loop(acc, fun, ref, monitor, fail_safe_timeout)
 
         case result do
-          {:ok, _} ->
+          {:ok, {status, headers, _}} ->
+            metadata = Map.merge(metadata, %status: status, headers: headers})
```
Comment (sneako) [inline lib/finch/http2/pool.ex:62]:
```suggestion
            metadata = Map.merge(metadata, %{status: status, headers: headers})
```

**Pull request merged**